lib/modules/manager/gradle/parser.ts

```diff
@@ -396,6 +397,19 @@ function processLibraryDep(input: SyntaxHandlerInput): SyntaxHandlerOutput {
   return res;
 }
 
+function processApplyFrom({
+  tokenMap,
+  variables,
+}: SyntaxHandlerInput): SyntaxHandlerOutput {
+  let scriptFile: string | null = tokenMap.scriptFile?.value;
```

Comment:
```suggestion
  let scriptFile = tokenMap.scriptFile?.value ?? null;
```
don't silently use undefined as null.

Reply:
Doesn't work:
```
parser.ts:408:5 - error TS2322: Type 'string | null' is not assignable to type 'string'.
Type 'null' is not assignable to type 'string'.

408     scriptFile = interpolateString(token.children, variables);
```

It seems that with `let scriptFile = tokenMap.scriptFile?.value ?? null;` the compiler assumes that `scriptFile` needs to be a string.

Reply:
Fixed in [c092a5d](https://github.com/renovatebot/renovate/pull/16030/commits/c092a5d3ba87df44afa5f006b03c2039e2311b36).